dataland-frontend/src/components/resources/frameworkDataSearch/lksg/LksgPanel.vue

```diff
@@ -186,6 +198,71 @@ export default defineComponent({
         : getCountryNameFromCountryCode(kpiValue as string) ?? kpiValue;
     },
 
+    /**
+     * Generates a list of readible strings (or just a single one) combining suppliers and their associated countries
+     * @param numberOfSuppliersPerCountryCode the map of number of suppliers and associated companies
+     * from which strings are written
+     * @returns the constructed collection of readable strings
+     */
+    generateReadableCombinationOfNumberOfSuppliersAndCountries(
+      numberOfSuppliersPerCountryCode?: Map<string, number | undefined | null>
+    ) {
+      if (numberOfSuppliersPerCountryCode != undefined) {
+        const readableListOfSuppliersAndCountries = Array.from(numberOfSuppliersPerCountryCode.entries()).map(
+          ([countryCode, numberOfSuppliers]) => {
+            const countryName = getCountryNameFromCountryCode(countryCode) ?? countryCode;
+            if (numberOfSuppliers != undefined) {
+              return String(numberOfSuppliers) + " suppliers from " + countryName;
+            } else {
+              return "There are suppliers from " + countryName;
+            }
+          }
+        );
+        if (readableListOfSuppliersAndCountries.length > 1) {
+          return readableListOfSuppliersAndCountries;
+        } else {
+          return readableListOfSuppliersAndCountries[0];
+        }
+      } else {
+        return null;
+      }
+    },
+
+    /**
+     * Converts the map of ProcurementCategory and LksgProductCategory into an array for a proper handling of the
+     * DetailsCompanyDataTable in the LksgCompanyDataTable (modal showing information related to Procurement Categories)
+     * @param inputObject Map to convert to array
+     * @returns The constructed map
+     */
+    reformatProcurementCategoriesValue(inputObject: Map<ProcurementCategoryType, LksgProcurementCategory> | null) {
+      if (inputObject == null) return null;
+      const listOfProductCategories = [];
+      for (const [procurementCategory, lksgProductCategory] of Object.entries(inputObject)) {
```

Comment:
I think you can give the variables a type in the for loop.  Then you don't need to do "as LksgProcurementCategory" so often in the following code.

Reply:
Somehow in my IDE, it does not recognize that `lksgProcurementCategory` is of type `LksgProcurementCategory`.
I have no clue why:

![image](https://github.com/d-fine/Dataland/assets/99255296/d26f6f4d-cd6c-4c8e-995b-49f710f11822)

I am quite confused.  Does your IDE also show warnings here?

Reply:
Nope. It is properly identified. Weird.

![image](https://github.com/d-fine/Dataland/assets/131248192/1850b141-a5a3-4b3a-9310-392af17678b9)

Reply:
I think the problem is that IntelliJ doesn't know what `lksgProcurementCategory` is, even though it is defined in the map(). It shows me as undefined:

<img width="584" alt="image" src="https://github.com/d-fine/Dataland/assets/131248192/d78d7caf-86e2-466d-a417-9b787f42b4b2">

Reply:
This works:
<img width="725" alt="image" src="https://github.com/d-fine/Dataland/assets/131248192/8b5702a2-f075-453f-aa47-23a29b2ae44e">

Reply:
Pushed the above solution (but named `inputEntry`). Not too elegant, but solves the issue.
Done.

Reply:
I've further improved this code:
```
const [procurementCategoryType, lksgProcurementCategory] = inputEntry;
```
... instead of splitting it using entry[0] and entry[1]

Reply:
Great, I like that and my IDE is also happy.
I am closing this

👍  thx